# Example 2

Методом максимального правдоподобия найти точечную оценку параметра λ по данной выборке
```
Х: 1-3| 3-5| 5-7| 7-9| 9-11| 11-13| 13-15| 15-17| 17-19
n: 5  | 6  | 7  | 15 | 22  | 27   | 30   | 34   | 35
```
при условии, что соответствующая непрерывная случайная величина имеет плотность распределения f(x)=λexp(λ(x−20)),x≤20.


In [1]:
import pandas as pd

In [2]:
df = pd.DataFrame({"X":[[1,2,3], [3,4,5], [5,6,7], [7,8,9], [9,10,11], [11,12,13], [13,14,15], [15,16,17], [17,18,19]],
                  "n": [5, 6, 7, 15, 22, 27, 30, 34, 35]})
df

X   n
0     [1, 2, 3]   5
1     [3, 4, 5]   6
2     [5, 6, 7]   7
3     [7, 8, 9]  15
4   [9, 10, 11]  22
5  [11, 12, 13]  27
6  [13, 14, 15]  30
7  [15, 16, 17]  34
8  [17, 18, 19]  35

при условии, что соответствующая непрерывная случайная величина имеет плотность распределения
$ f(x) = \begin{cases} \lambda \exp[\lambda(x-20)], x \leq 20 \\ 0 , x > 20 \end{cases}$

Всего провели *n* испытаний и разбили их на *k* классов

In [4]:
n = df.n.sum()
n

181

In [6]:
k = df.X.size
k

9

ф-я правдоподобия имеет следующий вид $L = \prod^{181}_{i=1} f(x_{i}) = \lambda^{181} e^{\lambda \sum^{181}_{i=1}x_{i} - 3620\lambda}$

Найдем логарифмическую функцию правдоподобия: $\ln L = 181 \ln \lambda + \lambda \sum^{181}_{i=1}x_{i} - 3620\lambda$

Найдем первую производную по $\lambda$ : $\dfrac{d\ln L}{d \lambda} = \dfrac{181}{\lambda} + \sum^{181}_{i=1}x_{i} - 3620$ 

Напишем уравнение правдоподобия, для чего приравняем первую производную к нулю: $\dfrac{181}{\lambda} + \sum^{181}_{i=1}x_{i} - 3620 = 0$

Найдем критическую точку, для чего решим полученное уравнение относительно $\lambda$: $\lambda = \dfrac{1}{20 - \overline{x_{B}}}$

Найдем вторую производную по $\lambda$: $\dfrac{d^{2}\ln L}{d \lambda^{2}} = - \dfrac{181}{\lambda^{2}}$

вторая производная отрицательная следовательно $\lambda = \dfrac{1}{20 - \overline{x_{B}}}$ точка максимума и, значит, в качестве оценки
максимального правдоподобия параметра  $\lambda$ надо принять величину $\dfrac{1}{20 - \overline{x_{B}}}$ 

в качестве $x_{i}$ среднее значение в интервале

In [18]:
df["x_in_i"] = [df.X[i][1] * df.n[i] for i in df.index ]

In [19]:
df

X   n  x_in_i
0     [1, 2, 3]   5      10
1     [3, 4, 5]   6      24
2     [5, 6, 7]   7      42
3     [7, 8, 9]  15     120
4   [9, 10, 11]  22     220
5  [11, 12, 13]  27     324
6  [13, 14, 15]  30     420
7  [15, 16, 17]  34     544
8  [17, 18, 19]  35     630

In [20]:
x_B = 1 / n * df.x_in_i.sum()
x_B

12.895027624309392

In [21]:
lambda_ = 1 / (20 - x_B)
lambda_

0.14074650077760498

точечную оценку параметра $\lambda \approx 0,141$